packages/next-dashboard/src/components/layout/Main.js

```diff
@@ -34,39 +34,105 @@ function DashboardLayout({
   children,
   id,
   contentContainerWidth,
-  header,
+  header: propHeader,
   sidebar,
   footer,
 }: Props): React$Element<typeof LayoutContext.Provider> {
   const ctx = useContext(DashboardContext);
   const { [LAYOUT]: ctxProps } = ctx;
-  const lctx = useCreateLayoutContext(
-    ctxProps.initialState,
-    ctxProps.persist,
-    ctx,
-  );
+  const lctx = useCreateLayoutContext(ctxProps.initialState, ctxProps.persist);
   const {
     getState,
     theme: { class: themeClass },
     modalActive,
   } = lctx;
 
+  const isFullscreen = lctx.getTemp('isFullscreen', false);
+  const isFullscreenMoving = lctx.getTemp('fullscreen-cursor-moving', false);
```

Comment:
So what exactly is `getTemp` meant to be anyway? Is it something meant to be removed later on? in which sense is it temporary? (assuming that's the sense of 'Temp' here) Are 'temps' properties set on the layout context?

Reply:
getTemp is temporary state that only lives in the current Layout, as in, when it gets remade on, say a page reload, or recreated in a new page component, it gets cleared.

Reply:
As opposed to the persistent nature of getState/setState, which survives things like reloads and navigation

Reply:
Oh okay!